src/inc/opcodes.h

```diff
@@ -601,7 +605,8 @@ STARTOP(put_subs)
     if (slot)
     {
         int index = seg.findClassIndex(input_class, slot->gid());
-        is->setGlyph(&seg, seg.getClassGlyph(output_class, index));
+        asset(index >=0 );
+        is->setGlyph(&seg, seg.getClassGlyph(output_class, static_cast<uint16>(index)));
```

Comment:
The question I would ask here for the sake of robustness: What does actually getClassGlyph return when called with an index like 31767 (from -1), and how does setGlyph react to getting fed that return value?

Reply:
My initial intention of the code was limited to reducing warnings on locally building LibreOffice,which relies on this component, and haven't researched what would happen in that case, but making use of the opportunity of being asked, I tried to understand terms used in the code, like font,face,segment, slif , GDL and more. However, the links to useful information on README of this project were broken. Sorry, I'm currently at a loss.

For now, I found that I typo-ed "assert" as "asset". I will fix that.

Reply:
no assert!

Reply:
If getClassGlyph is called with a an index that is > than the limit it can handle, it returns 0 (and since the index is uint, it is never < 0). This gives us an adequate garbage in, garbage out error handling. The Graphite engine is not in an a position to return errors, it has to do something with whatever is thrown at it, even if the results are rubbish.

Reply:
Is perhaps the value or index 0 itself also classified as such a special reference value, or an error condition? The last comment seems to suggest so. E.g. the index 0 could indeed have been utilized as a valid but special index to hold, say, the "substitute glyph" to use as fallback or during "silent recovery" whenever the libray is thrown some bad input.